src/main/java/io/fabric8/vertx/maven/plugin/utils/ScmSpy.java

```diff
@@ -0,0 +1,96 @@
+ * @author kameshs
+ */
+public class ScmSpy {
+
+    private static String DEFAULT_DATE_PATTERN = "yyyyMMdd HH:mm:ss z";
+
+    private final ScmManager scmManager;
+    private final SimpleDateFormat simpleDateFormat;
+
+    public ScmSpy(ScmManager scmManager) {
+        this.scmManager = scmManager;
+        simpleDateFormat = new SimpleDateFormat(DEFAULT_DATE_PATTERN);
+    }
+
+    /**
+     * The method that is used to determine the SCM type based on the SCM url pattern
+     * @see <a href="https://maven.apache.org/components/scm/scm-url-format.html">Maven SCM URL Format</a>
+     * @param scmUrl - the SCM url that needs to be parsed to determine the SCM type, the value of url is typically the
+     *               "connection" or "developerConnection" element value of the maven <scm/>
+     * @return String - the SCM type, e.g. for git it will be "git" , for subversion it will be "svn"
+     * @throws ScmRepositoryException - any SCM related exceptions that might happen when checking the SCM type
+     * @throws NoSuchScmProviderException - if an invalid url is given as part of maven <scm> "connection" or "developerConnection"
+     */
+    public String getScmType(String scmUrl) throws ScmRepositoryException, NoSuchScmProviderException {
+        return ScmUrlUtils.getProvider(scmUrl);
+    }
+
+    /**
+     * This method extracts the simple metadata such as revision, timestamp of the commit/hash, author of the commit
+     * from the Changelog available in the SCM repository
+     * @param scmUrl - the SCM url to get the SCM connection
+     * @param workingDir - the Working Copy or Directory of the SCM repo
+     * @return a {@link Map<String,String>} of values extracted form the changelog, with Keys from {@link ExtraManifestKeys}
+     * @throws IOException - any error that might occur while manipulating the SCM Changelog
+     * @throws ScmException - other SCM related exceptions
+     */
+    public Map<String, String> getChangeLog(String scmUrl, File workingDir) throws IOException, ScmException {
+        Map<String, String> changeLogMap = new HashMap<>();
+        ScmRepository scmRepository = scmManager.makeScmRepository(scmUrl);
+        ChangeLogScmResult scmResult = scmManager.changeLog(new ChangeLogScmRequest(scmRepository,
+            new ScmFileSet(workingDir, "*")));
+        if (scmResult.isSuccess()) {
+            List<ChangeSet> changeSetList = scmResult.getChangeLog().getChangeSets();
+            if (changeSetList != null && !changeSetList.isEmpty()) {
+                //get the latest changelog
+                ChangeSet changeSet = changeSetList.get(0);
+                changeLogMap.put(ExtraManifestKeys.scmType.name(), getScmType(scmUrl));
+                changeLogMap.put(ExtraManifestKeys.scmRevision.name(), changeSet.getRevision());
+                changeLogMap.put(ExtraManifestKeys.timestamp.name(), simpleDateFormat.format(changeSet.getDate()));
```

Comment:
yeah we can add, as current timestamp as as `Build-Timestamp` and the other timestamp as `Last-Commit-Timestamp`. ok ?